### PR TITLE
fix: allow submodels to overwrite extra field info

### LINKED ISSUE #3934: sub-model cannot always re-write field infos from the same BaseModel.
### Checks

* [-] I added a descriptive title to this issue
* [-] I have searched (google, github) for similar issues and couldn't find anything
* [-] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.9.0
            pydantic compiled: True
                 install path: /home/zuobinhua/anaconda3/lib/python3.7/site-packages/pydantic
               python version: 3.7.6 (default, Jan  8 2020, 19:59:22)  [GCC 7.3.0]
                     platform: Linux-4.15.0-159-generic-x86_64-with-debian-buster-sid
     optional deps. installed: ['email-validator', 'typing-extensions']
```
When creating two sub models(A, B) from Pydantic BaseModel, with fields infos defined in Config. Then creating another two sub models(Model1, Model2) from A and B, trying to rewrite the extra keys (we used `level` in the following code) in field info. Only one of the sub models can re-write the extra value correctly. 
As shown in the following code, `extra` in m2.__fields__["root"].field_info got level=3, but for m1 it still got level=1.

```py
from typing import Optional
from pydantic import BaseModel
from pydantic import Field


class A(BaseModel):
    root: Optional[str]
    class Config:
        fields = {
            "root": {"description": "root path of data", "level": 1},
        }

class B(BaseModel):
    root: Optional[str]
    class Config:
        field = {
            "root": {"description": "root path of data", "level": 1},
        }

class Model1(A):
    root: str = Field("asa", description="image height", level=3)

class Model2(B):
    root: str = Field("asa", description="image height", level=3)

m1 = Model1()
print(m1.__fields__["root"].field_info)
# default='asa' description='image height' extra={'level': 1}

m2 = Model2()
print(m2.__fields__["root"].field_info)
# default='asa' description='image height' extra={'level': 3}

```

### PR DESCRIPTION
**Pull request by PrettyWood**
## Change Summary
When creating a submodel, it was not possible to overwrite the "extra" part of the field info

## Related issue number
fix #3934 

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by PrettyWood** [comment]
please review

File: docs/requirements.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ hypothesis==6.31.6
 markdown-include==0.6.0
 mdx-truly-sane-lists==1.2
 mkdocs==1.2.3
+jinja2<3.1  # https://github.com/mkdocs/mkdocs/pull/2795 and https://github.com/pallets/jinja/pull/1544
```
Comment (hramezani) [inline docs/requirements.txt:8]: @PrettyWood Do we still need this?

**Comment by hramezani** [comment]
@PrettyWood I think you just need to update your patch and rebase.
Other than that LGTM

**Comment by hramezani** [comment]
please update

File: docs/requirements.txt (hunk 1 above)
Comment (PrettyWood) [reply docs/requirements.txt]: Nope. I removed it on merge

**Comment by PrettyWood** [comment]
Thanks for the review @hramezani

**Comment by PrettyWood** [comment]
please review 🙏

**Review by hramezani — APPROVED** [review]
Great :sunglasses:

**Comment by samuelcolvin** [comment]
thanks so much.

**Pull request merged**